jets/d/by_del.c

```diff
@@ -0,0 +1,92 @@
+/* j/4/by_del.c
+**
+*/
+#include "all.h"
+
+/* functions
+*/
+  u3_noun
+  u3qdb_del(u3_noun a,
+            u3_noun b)
+  {
+    if ( u3_nul == a ) {
+      return u3_nul;
+    }
+    else {
+      u3_noun l_a, n_a, r_a, pn_a, qn_a;
+      /* u3_noun d, l_d, n_d, r_d; */
+
+      if ( (c3n == u3r_trel(a, &n_a, &l_a, &r_a)) ||
+           (c3n == u3r_cell(n_a, &pn_a, &qn_a)) )
+      {
+        return u3m_bail(c3__exit);
+      }
+      else if ( c3n == u3r_sing(pn_a, b) ) {
+        if ( c3y == u3qc_gor(b, pn_a) ) {
+          return u3nt(u3k(n_a),
+                      u3qdb_del(l_a, b),
+                      u3k(r_a));
+        } else {
+          return u3nt(u3k(n_a),
+                      u3k(l_a),
+                      u3qdb_del(r_a, b));
+        }
+      }
+      else {
+        if ( u3_nul == l_a) {
+          return u3k(r_a);
+        } else if ( u3_nul == r_a) {
+          return u3k(l_a);
+        } else {
+          u3_noun n_l_a, l_l_a, r_l_a;
+          u3_noun n_r_a, l_r_a, r_r_a;
+
+          if ( (c3n == u3r_trel(l_a, &n_l_a, &l_l_a, &r_l_a) ) ||
+               (c3n == u3r_trel(r_a, &n_r_a, &l_r_a, &r_r_a) ) ) {
+            return u3m_bail(c3__exit);
+          } else {
+
+            if ( c3y == u3qc_vor(u3h(n_l_a), u3h(n_r_a)) ) {
```

Comment:
I think you should `(c3y == u3du(n_l_a)) && (c3y == u3du(n_r_a))` before this condition. It should bail %exit otherwise.